cheri_compressed_cap_common.h

```diff
@@ -67,22 +67,32 @@ _CC_STATIC_ASSERT_SAME(_CC_N(FIELD_INTERNAL_EXPONENT_SIZE) + _CC_N(FIELD_EXP_NON
                            _CC_N(FIELD_EXPONENT_LOW_PART_SIZE),
                        _CC_N(FIELD_EBT_SIZE));
 
+// In order to allow vector loads and store from memory we can optionally reverse the first two fields.
 struct _cc_N(cap) {
     /* offset = cursor - base */
+#ifdef _CC_REVERSE_PESBT_CURSOR_ORDER
+    /* Currently only updated lazily when compress is called.
+     * We might want to make this the canonical field and slice out other fields on demand.*/
+    uint64_t cached_pesbt;
     _cc_addr_t _cr_cursor; /* Capability cursor */
-    _cc_addr_t cr_base;    /* Capability base addr */
+#else
+    _cc_addr_t _cr_cursor;
+    _cc_addr_t cached_pesbt;
+#endif
     _cc_length_t _cr_top;  /* Capability top */
+    _cc_addr_t cr_base;    /* Capability base addr */
     uint32_t cr_perms;     /* Permissions */
     uint32_t cr_uperms;    /* User Permissions */
     uint32_t cr_otype;     /* Object Type, 24/16 bits */
     /* Original EBT from the decompressed capability. If you modify
      * cursor/base/top, you must be sure to either recalculate this field to
      * match or that the result is still representable, since this will be the
      * EBT written out to memory */
-    uint32_t cr_ebt;     /* Exponent/Base/Top */
     uint8_t cr_flags;    /* Flags */
     uint8_t cr_reserved; /* Remaining hardware-reserved bits to preserve */
     uint8_t cr_tag;      /* Tag */
+    uint8_t cr_bounds_valid /* Set if bounds decode was given an invalid cap */;
+    uint64_t padding;
```

Comment:
This is still there

Reply:
Padding is (hopefully this time really) gone. As far as cr_bounds_valid is concerned, the semantics of it are ifdef-ed away, but I imagine it is a useful debug bit to know if a capability has a valid encoding, so it may as well be kept.